### PR TITLE
Move Daml Profiler to EE version of sandbox/sandbox-classic

### PR DESCRIPTION
**Pull request by cocreature**
This splits Sandbox targets into EE/CE targets and exposes the option
in the EE version. The option still exists in the CE option for now
until we have released EE artifacts to not break users that might know
about it without an alternative.

There is also a small test that makes sure that this actually works
since classpaths are dumb and it didn’t work at first.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/daml-sdk/validate.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+# --- end runfiles.bash initialization v2 ---
+
+JAVA=$(rlocation "$TEST_WORKSPACE/$1")
+SDK_CE=$(rlocation "$TEST_WORKSPACE/$2")
+SDK_EE=$(rlocation "$TEST_WORKSPACE/$3")
+
+for cmd in sandbox sandbox-classic; do
+    ! $JAVA -jar $SDK_CE $cmd --help | grep -q profile-dir
```
Comment (garyverhaegen-da) [inline daml-assistant/daml-sdk/validate.sh:21]: Have you seen this fail? Without `-e` my understanding is that Bash will completely ignore all but the last iteration of the second loop.

File: ledger/sandbox-common/BUILD.bazel [hunk 2]
```diff
@@ -57,6 +57,47 @@ da_scala_library(
...
+    exports = [":sandbox-common"],
+    deps = [
+        ":sandbox-common",
+        "//daml-lf/data",
+        "//ledger/caching",
+        "//ledger/ledger-api-common",
+        "//ledger/participant-integration-api",
+        "//ledger/participant-state",
+        "//libs-scala/ports",
+    ],
```
Comment (garyverhaegen-da) [inline ledger/sandbox-common/BUILD.bazel:78]: Would it be worth extracting those deps to a variable? Do we expect ce & ee to diverge here?

Comment (garyverhaegen-da) [reply ledger/sandbox-common/BUILD.bazel:78]: Actually could this be turned into a loop like `sandbox-classic-{}` above?

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you!

File: ledger/sandbox-common/BUILD.bazel (hunk 2 above)
Comment (cocreature) [reply ledger/sandbox-common/BUILD.bazel:78]: They start diverging as soon as we remove the profiler option so yes, I think we can expect this.

Comment (garyverhaegen-da) [reply ledger/sandbox-common/BUILD.bazel:78]: Something like
```
for (edition, tags) in [("ee", []), ("ce", [...])]
```

Comment (cocreature) [reply ledger/sandbox-common/BUILD.bazel:78]: hm, I guess my reasoning was that there is no reason to believe that those two targets share anything apart from the name in general so I’d rather not tie them together artificially.

**Review by garyverhaegen-da — APPROVED** [review]

File: ledger/sandbox-common/BUILD.bazel (hunk 2 above)
Comment (garyverhaegen-da) [reply ledger/sandbox-common/BUILD.bazel:78]: 👍

File: daml-assistant/daml-sdk/validate.sh (hunk 1 above)
Comment (cocreature) [reply daml-assistant/daml-sdk/validate.sh:21]: so if I replace this with `exit 1` it actually does fail so I think `!` doesn’t do what I thought it does.

Comment (cocreature) [reply daml-assistant/daml-sdk/validate.sh:21]: ahrg apparently `!` negates `set -e`. I hate bash so much.

Comment (garyverhaegen-da) [reply daml-assistant/daml-sdk/validate.sh:21]: That copy-pasted segment is hard to parse. Maybe add an explicit `set -euo pipefail` after it? AFAIK `set` is idempotent. I'm also not quite sure about the priorities of `!` and `|`; it may be worth adding parentheses and/or moving the exclamation mark to the grep command.

Comment (garyverhaegen-da) [reply daml-assistant/daml-sdk/validate.sh]: You could expend a few more characters and just be explicit:
```
for cmd in sandbox sandbox-classic; do
    if ! $JAVA -jar $SDK_CE $cmd --help | grep -q profile-dir; then
        exit 1
    fi
done
```
